File: urbanairship-layout/src/main/java/com/urbanairship/android/layout/model/EmptyModel.java
/* Copyright Airship and Contributors */

package com.urbanairship.android.layout.model;

import com.urbanairship.android.layout.property.Border;
import com.urbanairship.android.layout.property.Color;
import com.urbanairship.android.layout.property.ViewType;
import com.urbanairship.android.layout.view.EmptyView;
import com.urbanairship.json.JsonException;
import com.urbanairship.json.JsonMap;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/**
 * An empty view that can have a background and border.
 *
 * Useful for the nub on a banner or as dividers between items.
 *
 * @see EmptyView
 */
public class EmptyModel extends BaseModel {
    public EmptyModel(@Nullable Color backgroundColor, @Nullable Border border) {
        super(ViewType.EMPTY_VIEW, backgroundColor, border);
    }

    @NonNull
    public static EmptyModel fromJson(@NonNull JsonMap json) throws JsonException {
        Color backgroundColor = backgroundColorFromJson(json);
        Border border = borderFromJson(json);
        return new EmptyModel(backgroundColor, border);
    }
}

File: urbanairship-layout/src/main/java/com/urbanairship/android/layout/model/EmptyModel.kt
/* Copyright Airship and Contributors */
package com.urbanairship.android.layout.model

import android.content.Context
import com.urbanairship.android.layout.environment.ModelEnvironment
import com.urbanairship.android.layout.environment.ViewEnvironment
import com.urbanairship.android.layout.info.EmptyInfo
import com.urbanairship.android.layout.view.EmptyView

/**
 * An empty view that can have a background and border.
 * Useful for the nub on a banner or as dividers between items.
 *
 * @see EmptyView
 */
internal class EmptyModel(
    viewInfo: EmptyInfo,
    environment: ModelEnvironment,
    properties: ModelProperties
) : BaseModel<EmptyView, EmptyInfo, BaseModel.Listener>(
    viewInfo = viewInfo,
    environment = environment,
    properties = properties
) {

    override fun onCreateView(
        context: Context,
        viewEnvironment: ViewEnvironment,
        itemProperties: ItemProperties?
    ) = EmptyView(context, this).apply {
        id = viewId
    }
}
